rmp/src/value.rs

```diff
@@ -481,31 +481,51 @@ impl From<usize> for Value {
 
 impl From<i8> for Value {
     fn from(v: i8) -> Value {
-        Value::Integer(Integer::I64(From::from(v)))
```

Comment:
Why have you changed this? I think the proper solution will be to check whether the incoming value fits in the shortest-bytes-representation requirement.